equinix/resource_fabric_cloud_router.go

```diff
@@ -207,187 +207,187 @@ func resourceFabricCloudRouter() *schema.Resource {
+	if projectTerraform == nil {
+		return nil
 	}
-	return mappedPr
+	var project *fabricv4.Project
+	projectMap := projectTerraform[0].(map[string]interface{})
+	projectId := projectMap["project_id"].(string)
+	project = &fabricv4.Project{ProjectId: projectId}
+
+	return project
 }
 func resourceFabricCloudRouterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	client := meta.(*config.Config).FabricClient
-	ctx = context.WithValue(ctx, v4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
+	client := meta.(*config.Config).NewFabricClientForSDK(d)
+	ctx = context.WithValue(ctx, fabricv4.ContextAccessToken, meta.(*config.Config).FabricAuthToken)
 	schemaNotifications := d.Get("notifications").([]interface{})
-	notifications := equinix_fabric_schema.NotificationsToFabric(schemaNotifications)
+	notifications := equinix_fabric_schema.NotificationsTerraformToGo(schemaNotifications)
 	schemaAccount := d.Get("account").(*schema.Set).List()
-	account := accountCloudRouterTerraToGo(schemaAccount)
+	account := accountCloudRouterTerraformToGo(schemaAccount)
 	schemaLocation := d.Get("location").(*schema.Set).List()
-	location := locationCloudRouterTerraToGo(schemaLocation)
-	project := v4.Project{}
+	location := locationCloudRouterTerraformToGo(schemaLocation)
+	var project *fabricv4.Project
 	schemaProject := d.Get("project").(*schema.Set).List()
 	if len(schemaProject) != 0 {
-		project = projectCloudRouterTerraToGo(schemaProject)
+		project = projectCloudRouterTerraformToGo(schemaProject)
 	}
 	schemaPackage := d.Get("package").(*schema.Set).List()
-	packages := packageCloudRouterTerraToGo(schemaPackage)
+	packages := packageCloudRouterTerraformToGo(schemaPackage)
 
-	createRequest := v4.CloudRouterPostRequest{
-		Name:          d.Get("name").(string),
-		Type_:         d.Get("type").(string),
-		Location:      &location,
+	type_ := fabricv4.CloudRouterPostRequestType(d.Get("type").(string))
+	createCloudRouterRequest := fabricv4.CloudRouterPostRequest{
+		Name:          d.Get("name").(*string),
+		Type:          &type_,
+		Location:      location,
 		Notifications: notifications,
-		Package_:      &packages,
-		Account:       &account,
-		Project:       &project,
+		Package:       packages,
+		Account:       account,
+		Project:       project,
 	}
 
-	if v, ok := d.GetOk("order"); ok {
-		order := equinix_fabric_schema.OrderToFabric(v.(*schema.Set).List())
-		createRequest.Order = &order
+	if orderTerraform, ok := d.GetOk("order"); ok {
+		order := equinix_fabric_schema.OrderTerraformToGo(orderTerraform.(*schema.Set).List())
+		createCloudRouterRequest.Order = order
 	}
 
-	fcr, _, err := client.CloudRoutersApi.CreateCloudRouter(ctx, createRequest)
+	fcr, _, err := client.CloudRoutersApi.CreateCloudRouter(ctx).CloudRouterPostRequest(createCloudRouterRequest).Execute()
 	if err != nil {
 		return diag.FromErr(equinix_errors.FormatFabricError(err))
 	}
-	d.SetId(fcr.Uuid)
+	d.SetId(*fcr.Uuid)
```

Comment:
I recommend using the `Get<Attr>()` functions instead; their behavior more closely matches the old code and will save you from some (unfortunately not all) pointer headaches
```suggestion
	d.SetId(fcr.GetUuid())
```

Reply:
Will do 👍